eZ Platform/Admin UI/Field Type Edit/Requirements.md

```diff
@@ -1,45 +1,35 @@
+- The solution should enable CRUD operations on fields in the exact same way, in both eZ Platform UI (cf. Hybrid application prototype), and in any website directly powered by eZ Platform Web Framework (as opposed to a decoupled website, simply using eZ Platform APIs)
+- The solution should enable CRUD operations on fields in a remote website / web application (in any programming language) using the REST API, and should minimize the UI logic that will have to be implemented for that (e.g. moving as much as possible of the UI logic in the APIs, such as validation rules, etc., or exposing the Web UI through the API)
 
-# Firm
+## Preferred requirements
 
-The prototype *must* for the following interfaces:
-* Twig based content editing
-* A remote website (e.g. Angular)
-* An App using REST API (e.g.React Native)
+- The solution should enable CRUD operations on fields in a non-web application (e.g. native mobile apps, digital signage, iTV...) using the REST API and should minimize the UI logic that will have to be implemented
 
-Provide examples that:
+### Candidates
 
-* Demonstrate how we can define input validation
-* Demonstrate how we can define input UI commonly
-    * Preferably ending with auto generating on the different platforms a working default
-* Demonstrate how we can override UI components where platform specific implementation would be more desirable
+The repository-form solution is a strong candidate for the Web UI as it solves the first firm requirement by design. Repository-form was, however, created as a complement of Platform UI, and this spike should validate (by examples) that it's a viable standalone solution, and should benchmark other solutions too.
 
+Other approaches (either complementary or alternative) might be good candidates too, such as:
 
-# Preferred
+- Exposing part of the UI via REST API calls (answers the 2nd requirement)
+- Using webcomponents instead of pure Twig-based UI (answers the 1st and 2nd requirements and reduces dependencies on Symfony for the future)
+- Evolving repository-form: based on repository forms, but adding additional capabilities to better define UI behaviors in the API
+...?
 
-The prototype *should*:
-* Demonstrate how Symfony forms (repo-forms) can be used
-* Identify logic that should be moved to the API
-* Demonstrate how we can refactor how to store data using native or custom (e.g. MongoDB or remote AWS RDS + S3)
-* Demonstrate how to make content searchable (e.g. how to define facets and sub-structures for search)
-* Demonstrate how to display content of the Field Type for a preview or example rendering
+## Outcome
 
-
-# Bonus
-
-The prototype *may*:
-
-* Demonstrate how we could simplify our API for adding a new Field Type today
+The spike allocated for this requirement should provide a final recommendation and a draft specification so that a rewrite of the edit UI of all default Field Types for 2.x can start.
```

Comment:
Can maybe start to add a section that we call "Possible Acceptance Tests" where we with @papelipe define scenarios we want to battle test the outcome with. in this case field type candidates to be implemented.